### PR TITLE
use wix-node-build as a build tool

### PR DESCRIPTION
**Pull request by ronami**

### CODE REVIEW

File: test/setup.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+function gloabalize(key, value) {
+  var windowExists = typeof window !== "undefined";
+  var globalExists = typeof GLOBAL !== "undefined";
+
+  if (windowExists)
+    window[key] = value;
+
+  if (globalExists)
+    GLOBAL[key] = value;
+}
```
Comment (yanivefraim) [inline test/setup.js]: I think this should be called mocha-setup.js? @ronami

File: package.json [hunk 2]
```diff
@@ -4,18 +4,22 @@
...
-    "test": "mocha --require spec/setup.js \"spec/**/*.spec.js\"",
-    "compile": ":",
-    "prepublish": "npm run compile && npm test",
-    "build":"npm run build-src && npm run build-test-kit",
-    "build-src":"babel src --out-dir lib --copy-files --source-maps",
-    "build-test-kit":"babel test-kit --out-dir test-kit-lib --copy-files --source-maps",
-    "release": "npm install @wix/wnpm-ci && wnpm-release -- --no-shrinkwrap",
+    "lint": "wix-node-build lint",
+    "test": "wix-node-build test",
+    "build": "wix-node-build build && build-storybook",
```
Comment (shahata) [inline package.json]: "build": "wix-node-build lint && wix-node-build build && build-storybook"

File: .gitignore [hunk 3]
```diff
@@ -3,3 +3,10 @@ node_modules
 npm-debug.log
 lib/
 test-kit-lib/
+dist
+target
+coverage
+typings
```
Comment (shahata) [inline .gitignore]: typings is not relevant here I think

File: package.json [hunk 4]
```diff
@@ -4,18 +4,22 @@
...
   "main": "./lib/index.js",
   "scripts": {
-    "test": "mocha --require spec/setup.js \"spec/**/*.spec.js\"",
-    "compile": ":",
-    "prepublish": "npm run compile && npm test",
-    "build":"npm run build-src && npm run build-test-kit",
-    "build-src":"babel src --out-dir lib --copy-files --source-maps",
-    "build-test-kit":"babel test-kit --out-dir test-kit-lib --copy-files --source-maps",
-    "release": "npm install @wix/wnpm-ci && wnpm-release -- --no-shrinkwrap",
+    "lint": "wix-node-build lint",
```
Comment (shahata) [inline package.json]: no need for this

File: package.json [hunk 5]
```diff
@@ -4,18 +4,22 @@
...
+    "test": "wix-node-build test",
+    "build": "wix-node-build build && build-storybook",
+    "start": "wix-node-build start & npm run storybook",
```
Comment (shahata) [inline package.json]: && or &?

Comment (ronami) [inline package.json]: & is used to run both simultaneously, they both watch for file changes so they don't end until watch is finished.

File: package.json [hunk 6]
```diff
@@ -2,20 +2,29 @@
   "name": "wix-style-react",
   "description": "wix-style-react",
   "version": "1.1.0",
-  "main": "./lib/index.js",
+  "main": "./src/index.js",
```
Comment (shahata) [inline package.json]: should point to dist

File: package.json [hunk 7]
```diff
@@ -2,20 +2,29 @@
...
-    "test": "mocha --require spec/setup.js \"spec/**/*.spec.js\"",
-    "compile": ":",
-    "prepublish": "npm run compile && npm test",
-    "build":"npm run build-src && npm run build-test-kit",
-    "build-src":"babel src --out-dir lib --copy-files --source-maps",
-    "build-test-kit":"babel test-kit --out-dir test-kit-lib --copy-files --source-maps",
-    "release": "npm install @wix/wnpm-ci && wnpm-release -- --no-shrinkwrap",
+    "test": "wix-node-build test",
+    "build": "wix-node-build lint && wix-node-build build && build-storybook",
+    "start": "wix-node-build start & npm run storybook",
```
Comment (shahata) [inline package.json]: & or &&?

File: package.json [hunk 8]
```diff
@@ -2,20 +2,29 @@
...
+    "release": "wix-node-build release && gh-pages-auto-release --dist=storybook-static",
     "storybook": "start-storybook -p 6006",
-    "lint": "eslint test/ src/",
-    "pretest": "npm run lint -s",
-    "build-storybook": "build-storybook",
-    "deploy-storybook": "storybook-to-ghpages"
+    "compile": ":",
+    "prepublish": "npm run compile && npm test"
```
Comment (shahata) [inline package.json]: remove `compile` and `prepublish` steps

File: package.json (hunk 6 above)
Comment (yanivefraim) [inline package.json]: are we sure it won't break current project? they will just have babel running twice?

File: package.json [hunk 9]
```diff
@@ -2,20 +2,29 @@
   "name": "wix-style-react",
   "description": "wix-style-react",
   "version": "1.1.0",
-  "main": "./lib/index.js",
+  "main": "./src/index.js",
+  "files": [
+    "dist",
+    "src",
+    "stories",
+    ".babelrc"
```
Comment (yanivefraim) [inline package.json]: we should have `testkit` here is well

File: package.json [hunk 10]
```diff
@@ -2,20 +2,29 @@
   "name": "wix-style-react",
   "description": "wix-style-react",
   "version": "1.1.0",
-  "main": "./lib/index.js",
+  "main": "./src/index.js",
+  "files": [
+    "dist",
+    "src",
+    "stories",
```
Comment (shahata) [inline package.json]: why do we need to include stories in npm?

File: package.json (hunk 6 above)
Comment (shahata) [inline package.json]: 1) there's no problem with running babel twice
2) no one imports the main anyway probably

**Comment by shahata** [comment]
@ronami you also need to update readme to point to dist/src instead of lib and to dist/testkit instead of test-kit-lib

File: package.json (hunk 8 above)
Comment (yanivefraim) [inline package.json]: Done

File: package.json (hunk 6 above)
Comment (yanivefraim) [inline package.json]: Done

File: package.json (hunk 10 above)
Comment (yanivefraim) [inline package.json]: removed

**Comment by yanivefraim** [comment]
@shahata - I went over all comments and fixed (the only one I left is & vs && inside 'start').

Can you please review again?

**Comment by shahata** [comment]
lgtm! @3LOK ready to merge?

**Comment by 3LOK** [comment]
@yanivefraim @shahata - What are we planning on doing with all these warnings from wix-node-build?
```
npm WARN deprecated minimatch@2.0.10: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated npmconf@2.1.2: this package has been reintegrated into npm and is now out of date with respect to npm
npm WARN deprecated minimatch@0.2.14: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated graceful-fs@1.2.3: graceful-fs v3.0.0 and before will fail on node releases >= v7.0. Please update to graceful-fs@^4.0.0 as soon as possible. Use 'npm ls graceful-fs' to find it in the tree.
npm WARN deprecated minimatch@0.3.0: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated to-iso-string@0.0.2: to-iso-string has been deprecated, use @segment/to-iso-string instead.
npm WARN deprecated jade@0.26.3: Jade has been renamed to pug, please install the latest version of pug instead of jade
npm WARN deprecated node-uuid@1.4.7: use uuid module instead
npm WARN deprecated tough-cookie@2.2.2: ReDoS vulnerability parsing Set-Cookie https://nodesecurity.io/advisories/130
npm WARN deprecated lodash@1.0.2: lodash@<3.0.0 is no longer maintained. Upgrade to lodash@^4.0.0.
```

**Comment by shahata** [comment]
I'll look into it. you see it in np install? at least some of messages come from snyk

**Comment by 3LOK** [comment]
yes, from npm install.

I recommend using [npm-check-updates](https://www.npmjs.com/package/npm-check-updates). I used it to fix these warnings in other projects.

**Pull request merged**